azurerm/resource_arm_monitor_metric_alert.go

```diff
@@ -37,16 +37,25 @@ func resourceArmMonitorMetricAlert() *schema.Resource {
 				ValidateFunc: validate.NoEmptyStrings,
 			},
 
+			"target_resource_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"target_resource_region": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
 			"resource_group_name": azure.SchemaResourceGroupName(),
 
-			// TODO: Multiple resource IDs (Remove MaxItems) support is missing in SDK
-			// Issue to track: https://github.com/Azure/azure-sdk-for-go/issues/2920
-			// But to prevent potential state migration in the future, let's stick to use Set now
 			"scopes": {
 				Type:     schema.TypeSet,
 				Required: true,
 				MinItems: 1,
-				MaxItems: 1,
+				ForceNew: true,
```

Comment:
we only need to recreate the resource in some occasions, is it possible to make this conditional?

Reply:
This is a breaking change? And I believe you can do this via CustomizeDiff but i'm not sure how exactly to do it.